tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/context/BMainInstance.java

```diff
@@ -51,6 +52,25 @@ public class BMainInstance implements BMain {
     private String agentArgs = "";
     private BalServer balServer;
 
+    private static class StreamGobbler extends Thread {
+        private InputStream inputStream;
+        private PrintStream printStream;
+
+        public StreamGobbler(InputStream inputStream, PrintStream printStream) {
+            this.inputStream = inputStream;
+            this.printStream = printStream;
+        }
+
+        @Override
+        public void run() {
+            Scanner sc = new Scanner(inputStream, StandardCharsets.UTF_8);
```

Comment:
Shall we use a BufferedReader instead of Scanner as it is more efficient as it buffers data.